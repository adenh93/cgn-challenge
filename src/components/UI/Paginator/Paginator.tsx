import React, { SFC } from "react";
import { Button, Box } from "@material-ui/core";
import { ArrowLeft, ArrowRight } from "@material-ui/icons";
import useStyles from "./styles";

export interface PaginatorProps {
  disableNext: boolean;
  disablePrev: boolean;
  onClickNext: () => void;
  onClickPrev: () => void;
}

const Paginator: SFC<PaginatorProps> = ({
  disableNext,
  disablePrev,
  onClickNext,
  onClickPrev
}) => {
  const { root } = useStyles();
  return (
    <div className={root}>
      <Box mr={1}>
        <Button
          variant="contained"
          color="secondary"
          size="large"
          disabled={disablePrev}
          onClick={onClickPrev}
          startIcon={<ArrowLeft />}
        >
          Prev
        </Button>
      </Box>
      <Button
        variant="contained"
        color="secondary"
        size="large"
        disabled={disableNext}
        onClick={onClickNext}
        endIcon={<ArrowRight />}
      >
        Next
      </Button>
    </div>
  );
};

export default Paginator;
