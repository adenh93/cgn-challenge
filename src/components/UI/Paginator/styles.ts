import { makeStyles, createStyles } from "@material-ui/styles";
import { Theme } from "@material-ui/core";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex"
    }
  })
);

export default useStyles;
