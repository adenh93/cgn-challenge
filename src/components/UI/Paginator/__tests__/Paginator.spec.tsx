import * as React from "react";
import { create } from "react-test-renderer";
import Paginator, { PaginatorProps } from "../Paginator";
import { Button } from "@material-ui/core";

const props: PaginatorProps = {
  disableNext: false,
  disablePrev: false,
  onClickNext: () => {},
  onClickPrev: () => {}
};

describe("Alert Component", () => {
  test("it matches the snapshot", () => {
    const component = create(<Paginator {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("the previous button is disabled when disablePrev prop is passed", () => {
    const { root } = create(<Paginator {...{ ...props, disablePrev: true }} />);
    const prevButton = root.findAllByType(Button)[0];
    expect(prevButton.props["disabled"]).toBe(true);
  });

  test("the next button is disabled when disableNext prop is passed", () => {
    const { root } = create(<Paginator {...{ ...props, disableNext: true }} />);
    const prevButton = root.findAllByType(Button)[1];
    expect(prevButton.props["disabled"]).toBe(true);
  });
});
