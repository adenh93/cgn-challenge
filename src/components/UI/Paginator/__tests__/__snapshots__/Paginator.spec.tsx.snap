// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component it matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeLarge"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 7l-5 5 5 5V7z"
            />
            <path
              d="M24 0v24H0V0h24z"
              fill="none"
            />
          </svg>
        </span>
        Prev
      </span>
    </button>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Next
      <span
        className="MuiButton-endIcon MuiButton-iconSizeLarge"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 17l5-5-5-5v10z"
          />
          <path
            d="M0 24V0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;
